http/server.go

```diff
@@ -140,6 +145,16 @@ func (h *handler) getRand(ctx context.Context, round uint64) ([]byte, error) {
 	// First see if we should get on the synchronized 'wait for next release' bandwagon.
 	block := false
 	h.pendingLk.RLock()
+	if h.pending == nil {
+		h.pendingLk.RUnlock()
+		h.pendingLk.Lock()
+		if h.pending == nil {
+			h.pending = make([]chan []byte, 0)
+			go h.Watch(h.context)
+		}
+		h.pendingLk.Unlock()
+		h.pendingLk.RLock()
```

Comment:
This kind of code would just be potentially much simpler by just requiring the write side of the lock from the get go, no ? I dont think the perf impact would be drastic to require the write until line 160...

Reply:
Other idea is to use `sync.Once` to initialize `pending`

Reply:
The nice thing here is that it means that the common pathway is only hitting the low-contention read lock.

@nikkolasg  - if you `Lock()` you have to `Unlock()`, not `RUnlock()`, so there'd be probably as much complexity in holding either read or write until 160 and then figuring out which way to unlock at that point.

sync.Once is a separate atomic check on every access, right? dunno if it's worth worrying about that. probably premature optimization 😅